modules/planning/traffic_rules/change_lane.cc

```diff
@@ -76,8 +76,9 @@ bool ChangeLane::FilterObstacles(ReferenceLineInfo* reference_line_info) {
   }
   if (first_guard_vehicle) {
     guard_obstacles_.push_back(first_guard_vehicle);
+    return true;
   }
-  return true;
+  return false;
```

Comment:
we shall still return TRUE if first_guard_vehicle not exit. 
when this function return FALSE, this rule (change lane) would fail at line 139. and therefore, it shall return FALSE only if fatal error occurs.

Reply:
Agree with jmtao. The first_guard_vehicle can be nullptr in some cases (which means there are no guard vehicles). For those cases, the CreateGuardObstacle() function has a nullptr checking logic later at line 85. For FilterObstacles() function itself, it should always return TRUE unless some fatal error occurs.

Reply:
Agree with jmtao. The first_guard_vehicle can be nullptr in some normal cases (which means there are no guard vehicles). For those cases, the ApplyRule() function will check if the guard_obstacles_ vector is empty or not at line 142. For FilterObstacles() function itself, it should always return TRUE unless some unexpected fatal error occurs in planning.

Reply:
Thanks for your comments @jmtao @JasonXu12. so I will close this PR.